core/orch_test.go

```diff
@@ -419,6 +419,63 @@ func TestRemoveFromRemoteTranscoders(t *testing.T) {
 	assert.Equal(remoteTranscoderList, emptyTList)
 }
 
+//Validates that selectTranscoder will shift all streams to another valid transcoder if transcoder fails.
+func TestRemoveFromRemoteTranscoders_StreamFailover(t *testing.T) {
+	rtm := NewRemoteTranscoderManager()
+	assert := assert.New(t)
+	capabilities := NewCapabilities(append(DefaultCapabilities(), OptionalCapabilities()...), []Capability{})
+
+	var transcoder_1 = &StubTranscoderServer{manager: rtm}
+	var transcoder_2 = &StubTranscoderServer{manager: rtm}
+
+	assert.Nil(rtm.liveTranscoders[transcoder_1])
+	assert.Empty(rtm.remoteTranscoders)
+
+	// register 2 transcoders, 8 stream capacity each
+	go func() { rtm.Manage(transcoder_1, 8, capabilities.ToNetCapabilities()) }()
+	time.Sleep(1 * time.Millisecond) // allow time for first stream to register
```

Comment:
Won't this test be flaky? I think depending on timing (especially in ms) can make this test sometime pass, sometimes not. I'd avoid having time-dependent tests.

Reply:
I copied the wait commands from another test.  I think they are not needed. I didn't see anything in my development that indicates they are required, so I can try removing them.

Reply:
Yeah, then I suggest removing them if not needed. Then, Do you need to run these functions above in separate goroutines?

Reply:
Yes, apparently the separate go routine is required in this case, if not done with the go routine, the test will timeout. I tried removing the time.sleep between them, but it caused test to move on without transcoder connected at all.

I pushed a commit to cleanup a few related lines.

Reply:
Hmm, THB, in my opinion, it's better to not have tested with `time.Sleep()`, because they will sometimes fail sometimes pass in our CI. So, if it's not possible (or easy) to write a test without sleep, then I suggest removing this test and just merging without it. @thomshutt wdyt?

The improvement to the code is already good, so should be safe to merge.